[KB-299] Future works (#211)
chhoumann/thesis-chemcam: report_thesis/src/sections/future_work.tex

\section{Future Work}\label{sec:future_work}

> The findings of this study open several avenues for future research.

Reviewer: ```suggestion
The findings of this study present several opportunities for future research.
```

> Firstly, for our data partitioning algorithm, described in Section~\ref{subsubsec:dataset_partitioning}, we noted that finding the optimal percentile value $p$ that minimizes extreme values in the test set, while maintaining its general representativeness, is important.

Reviewer: ```suggestion
Firstly, regarding our data partitioning algorithm detailed in Section~\ref{subsubsec:dataset_partitioning}, we observed the significance of identifying the optimal percentile value $p$. 
This value is crucial for minimizing extreme values in the test set while preserving its overall representativeness.
```

> Future work should consider methods of quantitatively assesing and finding this value.

Reviewer: ```suggestion
Future work should explore quantitative methods for determining this optimal value. 
```

> Such methods could include supplementary extreme value testing to the data partitioning algorithm, where after the primary evaluation, additional testing is conducted using a small, separate subset of extreme values to assess the model's performance on these critical cases.

Reviewer: ```suggestion
Such methods could involve incorporating supplementary extreme value testing into the data partitioning algorithm. 
After the primary evaluation, additional testing could be conducted using a small, separate subset of extreme values to assess the model's performance in these critical scenarios.
```

> For example, this could involve slightly reducing the percentile value $p$ and using the extreme values that fall within this reduced range to evaluate the model.

Reviewer: ```suggestion
For example, this might involve slightly reducing the percentile value $p$ and using the extreme values that fall within this reduced range to evaluate the model's effectiveness.
```

> Another point of interest is limited data availability.

Reviewer: ```suggestion
Another point of interest is the limited data availability.
```

> The small dataset size naturally limits the amount of how many extreme values are present.

Reviewer: ```suggestion
The small dataset size inherently restricts the number of extreme values present.
```

> These extreme values are an essential part of improving the model's generalizability, as they are the most challenging cases to predict.

Reviewer: ```suggestion
The small dataset size inherently restricts the number of extreme values present.
These extreme values are crucial for enhancing the model's generalizability, as they represent the most challenging cases to predict.
```

Firstly, regarding our data partitioning algorithm detailed in Section~\ref{subsubsec:dataset_partitioning}, we observed the significance of identifying the optimal percentile value $p$. 
This value is crucial for minimizing extreme values in the test set while preserving its overall representativeness.
Future work should explore quantitative methods for determining this optimal value.
Such methods could involve incorporating supplementary extreme value testing into the data partitioning algorithm. 
After the primary evaluation, additional testing could be conducted using a small, separate subset of extreme values to assess the model's performance in these critical scenarios.
For example, this might involve slightly reducing the percentile value $p$ and using the extreme values that fall within this reduced range to evaluate the model's effectiveness.

Another point of interest is the limited data availability.
The small dataset size naturally limits the amount of how many extreme values are present.
These extreme values are an essential part of improving the model's generalizability, as they are the most challenging cases to predict.

> Future work should investigate methods of augmenting the dataset with synthetic extreme value data to provide the model with more exposure to these cases during training.

Reviewer: ```suggestion
Future research should investigate methods for augmenting the dataset with synthetic extreme value data to provide the model with more exposure to these cases during training.
```

Future work should also consider further experimentation with the choices of base estimators and meta-learners.

> Our study highlighted that multiple model and preprocessor configurations perform well.

Reviewer: ```suggestion
Our study demonstrated that various model and preprocessor configurations perform well.
```

> However, determining which configurations and meta-learner is optimal for a given oxide is a challenging task.

Reviewer: ```suggestion
However, identifying the optimal configurations and meta-learner for a specific oxide remains a challenging task.
```

> In this study, we used a simple grouping to ensure diversity in our base estimator selection, chosen from the top-performing configurations.

Reviewer: ```suggestion
In this study, we used a simple grouping method to ensure diversity in our base estimator selection, choosing from the top-performing configurations.
```

> This approach could be improved upon by, for example, developing more advanced selection methods that consider the base estimators and meta-learners in conjunction. 

Reviewer: ```suggestion
This approach could be enhanced by developing more advanced selection methods that consider the interactions between base estimators and meta-learners.
```

> \section{Future Work}\label{sec:future_work}
> The findings of this study present several opportunities for future research.
> Firstly, regarding our data partitioning algorithm detailed in Section~\ref{subsubsec:dataset_partitioning}, we observed the significance of identifying the optimal percentile value $p$. 
> This value is crucial for minimizing extreme values in the test set while preserving its overall representativeness.
> Future work should explore quantitative methods for determining this optimal value.
> Such methods could involve incorporating supplementary extreme value testing into the data partitioning algorithm. 
> After the primary evaluation, additional testing could be conducted using a small, separate subset of extreme values to assess the model's performance in these critical scenarios.
> For example, this might involve slightly reducing the percentile value $p$ and using the extreme values that fall within this reduced range to evaluate the model's effectiveness.
> 
> Another point of interest is the limited data availability.
> The small dataset size inherently restricts the number of extreme values present.
> These extreme values are crucial for enhancing the model's generalizability, as they represent the most challenging cases to predict.
> Future research should investigate methods for augmenting the dataset with synthetic extreme value data to provide the model with more exposure to these cases during training.
> 
> Future work should also consider further experimentation with the choices of base estimators and meta-learners.
> Our study demonstrated that various model and preprocessor configurations perform well.
> However, identifying the optimal configurations and meta-learner for a specific oxide remains a challenging task.
> In this study, we used a simple grouping method to ensure diversity in our base estimator selection, choosing from the top-performing configurations.
> This approach could be improved upon by, for example, developing more advanced selection methods that consider the interactions between base estimators and meta-learners.

Reviewer 2: ```suggestion
\section{Future Work}\label{sec:future_work}
The findings of this study present several opportunities for future research.
Firstly, regarding our data partitioning algorithm detailed in Section~\ref{subsubsec:dataset_partitioning}, we observed the significance of identifying the optimal percentile value $p$. 
This value is crucial for minimizing extreme values in the test set while preserving its overall representativeness.
Future work should explore quantitative methods for determining this optimal value.

Another potential improvement to the validation and testing approach we delineate is incorporating supplementary extreme value testing after the preimary evaluation.
This type of testing could be conducted using a small, separate subset of extreme values to assess the model's performance in these critical scenarios.
For example, this might involve slightly reducing the percentile value $p$ and using the extreme values that fall within this reduced range to evaluate the model's effectiveness.

Tackling the challenges of limited data availability has proven important, as we mention throughout our report.
The small dataset size inherently restricts the number of extreme values present.
These extreme values are crucial for enhancing the model's generalizability, as they represent the most challenging cases to predict.
Future research could investigate methods for augmenting the dataset with synthetic data, including extreme values, to provide the model with more exposure to these cases during training.
This is a hard task, as it requires the production of synthethic data for a physics-based process.
We contemplate that some approximation may be sufficient, and could be used, for instance, as part of a transfer-learning project as inital training material.

Future work should also consider further experimentation with the choices of base estimators and meta-learners.
Our study demonstrated that various model and preprocessor configurations perform well.
However, identifying the optimal configurations and meta-learner for a specific oxide remains a challenging task.
In this study, we used a simple grouping method to ensure diversity in our base estimator selection, choosing from the top-performing configurations.
We also experimented briefly with a grid search approach, which could be examined further.
However, using a variation of the optimization framework we presented is likely to provide better trade-offs, as we have discussed.
```
